### PR TITLE
Add allwinner A64 support 

### PR DESCRIPTION
**Pull request by HonJEK**
… incompatible pointer type

Signed-off-by: John-Eric Kamps <john-eric.kamps@honeywell.com>

I fixed a problem to compile Mali GPU on Allwinner A64 with the latest 5.0 Linux kernel.

### CODE REVIEW

**Comment by HonJEK** [comment]
Hi. Could it be now merged? Because there are also the new defines for the Allwinner A64.

**Comment by mripard** [comment]
Is this supposed to fix an issue with 4.17 or 5.0?

An initialization from incompatible pointer types should cause a warning, not an error.

The last build of 4.17 on arm64 also points that out:
https://travis-ci.org/mripard/sunxi-mali/jobs/430768060

Please also move the A64 commit to a separate PR.

File: patches/r6p2/0021-Add-A64-support.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ 	{ /* sentinel */ },
+ };
+@@ -311,6 +313,7 @@
+ 	}
+ 
+ 	if (of_device_is_compatible(np, "allwinner,sun8i-h3-mali") ||
++	    of_device_is_compatible(np, "allwinner,sun50i-a64-mali")||
+ 	    of_device_is_compatible(np, "allwinner,sun50i-h5-mali"))
+ 		clk_set_rate(mali->core_clk, 576000000);
+ 	else if (of_device_is_compatible(np, "allwinner,sun7i-a20-mali") ||
```
Comment (mripard) [inline patches/r6p2/0021-Add-A64-support.patch:26]: This should be added to the patch 5, not to a separate patch

File: patches/r6p2/series [hunk 2]
```diff
@@ -17,3 +17,4 @@ r6p2/0016-mali-support-building-against-4.16.patch
 0018-mali-support-building-against-4.20.patch
 0019-mali-support-building-against-5.0.patch
 r6p2/0020-mali-support-building-against-4.17.patch
+r6p2/0021-Add-A64-support.patch
```
Comment (mripard) [inline patches/r6p2/series]: And that way, it would apply to all the versions

**Review by mripard — CHANGES_REQUESTED** [review]

File: patches/r6p2/series (hunk 2 above)
Comment (HonJEK) [reply patches/r6p2/series]: OK. I was not aware about that that. Thanks

File: patches/r6p2/0021-Add-A64-support.patch (hunk 1 above)
Comment (HonJEK) [reply patches/r6p2/0021-Add-A64-support.patch:26]: Done. I was thinking to use a new patch file. OK

**Comment by HonJEK** [comment]
Could be closed as the change for building it against 4.17 has done from an other person and it is merged in master.
For the A64 change I created a new request. Thanks

**Pull request closed without merging**